tmt/steps/execute/internal.py

```diff
@@ -100,12 +110,14 @@ def _show_progress(self, progress, test_name, finish=False):
             self._previous_progress_message = ""
         sys.stdout.flush()
 
-    def _test_environment(self, test, extra_environment):
+    def _test_environment(self, test: Test, extra_environment: EnvironmentType) -> EnvironmentType:
```

Comment:
And `extra_environment` might be set to `None`, by `self._run_tests(self, guest, extra_environment=None)`, therefore `Optional[EnvironmentType] = None` & `extra_environment = extra_environment or {}` is needed.